components/omega/doc/design/PGrad.md

```diff
@@ -0,0 +1,196 @@
+
+$$ \frac{D \mathbf{u}_h}{D t } = \left( \frac{\partial}{\partial t} \right)_A  + \mathbf{u}_h\cdot \nabla_A + \omega\frac{\partial}{\partial A}, $$
+
+where $\omega$ is the cross coordinate flow.
+In the layered non-Boussinesq equations, the prognostic variable is the pressure thickness $h_k$, so that the geometric thickness (in meters) is a diagnostic variable defined as:
+
+$$ \Delta z_k = v_k h_k. $$
+
+The pressure at vertical cell interfaces is the found by summing the pressure thicknesses:
+
+$$ p_{K+1/2} = p_{surf} + g\sum_{k=1}^K h_k. $$
+
+The geopotential at vertical cell integrates is found by summing the pressure thicknesses multiplied by the specific volume:
+
+$$ \phi_{K+1/2} = g\left(z_b + \sum_{k=K}^{N} v_k h_k\right), $$
+
+where $z_b$ is the (positive-up) bottom depth.
+
+The discrete gradient operator at an edge is:
+
+$$ \nabla {(\cdot)} = \frac{1}{d_e} \sum_{i\in CE(e)} -n_{e,i} (\cdot)_i $$
+
+where $d_e$ is the distance between cell centers, $CE(e)$ are the cells on edge $e$, and $n_{e,i}$ is the sign of the edge normal with respect to cell $i$.
+Therefore the centered pressure gradient will be calculated as:
+
+$$ T^p_k = \frac{1}{d_e} \left( \widehat{v}_{k,e} \sum_{i \in CE(e)} n_{e,i} \overline{p}_{k,i} + \sum_{i\in CE(e)} n_{e,i} \overline{\phi}_{k,i}\right), $$
+
+$$ = \frac{1}{d_e} \left(  \sum_{i \in CE(e)} n_{e,i} \left( \widehat{v}_{k,e}\overline{p}_{k,i} + \overline{\phi}_{k,i} \right) \right), $$
+
+with the vertical averaging operator defined as:
+
+$$ \overline{(\cdot)} = \frac{1}{2} \left((\cdot)_{k+1/2} + (\cdot)_{k-1/2} \right)$$
+
+and the horizontal averaging operator:
+
+$$\widehat{(\cdot)} = \frac{1}{2} \left( (\cdot)_{i=1} + (\cdot)_{i=2}\right)$$
+
+## 4 Design
+
+### 4.1 Data types and parameters
+
+The `PressureGradient` class will be used to perform the horizontal gradients of the pressure and geopotential
+```c++
+class PressureGrad{
+    public:
+    private:
+        PressureGradCentered CenteredPGrad;
+        PressureGradHighOrder HighOrderPGrad;
+        PressureGradType PressureGradChoice;
+        I4 NVertLevels;
+        I4 NChuncks;
+        Array2DI4 CellsOnEdge;
+        Array1DReal DvEdge;
+        Array1DReal EdgeSignOnCell;
+}
+```
+The user will select a pressure gradient option at runtime in the input yaml file under the pressure gradient section:
+```yaml
+    PressureGrad:
+       PressureGradType: 'centered'
+```
+An `enum class` will be used to specify options for the pressure gradient used for an Omega simulation:
+```c++
+enum class PressureGradType{
+   Centered,
+   HighOrder
+}
+```
+The functions to compute the centered and high order pressure gradient terms will be implemented as functors and the pressure gradient class will have private instances of these classes.
+### 4.2 Methods
+
+```c++
+class PressureGrad{
+    public:
+        static PressureGrad *create();
+        void computePressureGrad();
+        void computePressureGradBtr();
```

Comment:
The Btr function is not described anywhere?

Reply:
Thanks, I can take care of that.